packages/neo-one-monitor/src/Reporter.js

```diff
@@ -0,0 +1,165 @@
+/* @flow */
+import { interval } from 'rxjs/observable/interval';
+import { mergeScan } from 'rxjs/operators';
+import type { Subscription } from 'rxjs/Subscription';
+
+import { utils } from '@neo-one/utils';
+
+import BrowserLogger from './BrowserLogger';
+import BrowserMetricsFactory from './BrowserMetricsFactory';
+import type { Report } from './types';
+
+import { KNOWN_LABELS, type CounterMetric } from './MonitorBase';
+
+const MAX_BACKLOG = 1000;
+
+export default class Reporter {
+  _endpoint: string;
+  _subscription: Subscription;
+  _requestCounter: CounterMetric;
+
+  constructor({
+    logger,
+    metricsFactory,
+    timer,
+    endpoint,
+  }: {|
+    logger: BrowserLogger,
+    metricsFactory: BrowserMetricsFactory,
+    timer: number,
+    endpoint: string,
+  |}) {
+    this._endpoint = endpoint;
+    this._requestCounter = metricsFactory.createCounter({
+      name: 'browser_request_counter',
+      help: 'Counts fetch requests from Browser Reporter to endpoint',
+      labelNames: [KNOWN_LABELS.ERROR],
+    });
+    const emptyBackReport = {
```

Comment:
nit: make this a constant at the module scope so you don't have to pass it in to `collectReport`